bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ChannelTypeBuilder.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.thing.type;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.types.EventDescription;
+import org.eclipse.smarthome.core.types.StateDescription;
+
+/**
+ * Builder for {@link ChannelType}s
+ *
+ * @author Stefan Triller- Initial contribution
+ *
+ */
+@NonNullByDefault
+public class ChannelTypeBuilder {
```

Comment:
Nice. We can reuse it in https://github.com/eclipse/smarthome/pull/5751/files#diff-77a713cacdb00eca969751106f87ceafR104.

What do you think about a second constructor?
```
public ChannelTypeBuilder(ChannelTypeUID channelTypeUID, String label) {
    this(channelTypeUID.getBindingId(), channelTypeUID.getId(), label);
}
```